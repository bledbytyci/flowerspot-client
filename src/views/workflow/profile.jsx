import AuthActionCreators from '../../auth/authActionCreators.js';
import AuthSelectors from '../../auth/authSelectors.js';
import React, {Component} from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import User from '../../user/user.js';
import Modal from '../common/modal.jsx';
import '../../styles/modal.css';
import ProfileForm from './forms/user/profileForm.jsx';
import { withRouter } from 'react-router-dom';

const mapStoreToProps = store => {
	return {
		user: AuthSelectors.getUser(store)
	}
}

const mapDispatchToProps = dispatch => {
	return {
		getProfile: () => dispatch(AuthActionCreators.getProfile()),
		logOutUser: () => dispatch(AuthActionCreators.logOutUser())
	}
}

export class Profile extends Component {
	constructor(props){
		super(props);
	}
	
	static get propTypes() {
		return {
			logOutUser: PropTypes.func,
			getProfile: PropTypes.func,
			user: PropTypes.instanceOf(User),
			match: PropTypes.object,
			history: PropTypes.object
		}
	}

	componentDidMount() {
		this.props.getProfile();
	}

	_onSave = () => {
		const {history, logOutUser} = this.props;
		logOutUser();
		history.push('/')
	}

	render(){
		const { match, user, history } = this.props;
		const modalProps = {
			title: '',
			show: match?.params?.show ? true : false,
			onHide: () => history.push('/'),
			width: 600
		}
		
		return (
			<Modal {...modalProps}>
				<ProfileForm user={user} onSave={this._onSave} />
			</Modal>

		)
	}
}

export default withRouter(connect(mapStoreToProps, mapDispatchToProps)(Profile));