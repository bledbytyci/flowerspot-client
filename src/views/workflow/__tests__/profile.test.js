import React from 'react';
import { Profile } from '../profile.jsx';
import { shallow} from 'enzyme';
import User from '../../../user/user.js';
import toJson from 'enzyme-to-json';

jest.unmock('../profile.jsx');
jest.unmock('../../../user/user.js');

describe('Profile component', () => {
    it('component is being rendered', () => {
		const props = {
			user: new User(),
			getProfile: jest.fn()
		}
		const profileComponent = shallow(<Profile {...props} />);
        expect(profileComponent).not.toBe(null);
		expect(props.getProfile).toBeCalled();
	});
});

describe('snapshots', () => {
	it('App snapshot', () => {
		const props = {
			user: new User(),
			getProfile: jest.fn()
		}
		const profileComponent = shallow(<Profile {...props} />);
		expect(toJson(profileComponent)).toMatchSnapshot();
	})
})
