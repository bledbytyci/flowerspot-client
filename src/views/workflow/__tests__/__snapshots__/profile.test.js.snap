// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshot 1`] = `
<Modal
  onHide={[Function]}
  show={false}
  title=""
  width={600}
>
  <ProfileForm
    onSave={[Function]}
    user={
      User {
        "date_of_birth": null,
        "email": "",
        "first_name": "",
        "id": "",
        "last_name": "",
        "password": "",
      }
    }
  />
</Modal>
`;
